### PR TITLE
MapGeneration

### PR DESCRIPTION
**Pull request by LeoWikstrom**
Maps are now procedurally generated. Try starting the game a number of times and see the different levels generated. THE ROOMS THEMSELVES ARE NOT A PART OF THIS PULL REQUEST, DO NOT REVIEW THEM HERE!

To test, comment line 110 in main.cpp:
//enemyFactory.Update(updateRate);

comment line 144 and uncomment line 145:
    //al->LoadMap(scene, "../Vendor/Resources/FirstMap.map");
    al->GenerateMap(scene, "../Vendor/Resources/Rooms");

### CODE REVIEW

**Review by AntonAsbrink — DISMISSED** [review]

**Review by AntonAsbrink — DISMISSED** [review]

File: StortSpelProjekt/Engine/src/Renderer/HLSL/ForwardPixel.hlsl [hunk 1]
```diff
@@ -92,7 +92,7 @@ PS_OUTPUT PS_main(VS_OUT input)
 			baseReflectivity);
 	}
 	
-	float3 ambient = float3(0.004f, 0.004f, 0.004f) * albedo;
+	float3 ambient = float3(0.05f, 0.05f, 0.05f) * albedo;
 	finalColor += ambient;
```
Comment (jocke1995) [inline StortSpelProjekt/Engine/src/Renderer/HLSL/ForwardPixel.hlsl]: Change this back to 0.004f. Fredrik told me to say that he changed this temporary and forgot to change them back.

File: StortSpelProjekt/Game/src/Gamefiles/Components/RangeComponent.cpp [hunk 2]
```diff
@@ -141,6 +141,8 @@ void component::RangeComponent::Attack()
 		pos.y = ParentPos.y + (forward.y / length);
 		pos.z = ParentPos.z + (forward.z / length) * (dim.z * scale.z / 2.0) * 1.1f;
 
+		Log::Print("Playerpos: %f, %f, %f\n", ParentPos.x, ParentPos.y, ParentPos.z);
```
Comment (jocke1995) [inline StortSpelProjekt/Game/src/Gamefiles/Components/RangeComponent.cpp]: remove this

**Review by jocke1995 — DISMISSED** [review]
A few questions about some stuff. Also, game is currently bugged, all the lights are in origo. If the old scene should be removed. You have to fix the teleport and enemies before this task can be added into game. Since right now, if we merge this, we wont have a game until another user story is completed.

**Review by AntonAsbrink — APPROVED** [review]

**Review by jocke1995 — DISMISSED** [review]
DirectionalLight in game is removed.
Particles lagging, but might not be your fault.. the search has begun

**Review by jocke1995 — APPROVED** [review]
Ok even if the dir light in "game" is removed. But since we're gonna change into the new maps like.. right away.. it won't matter as you said :)

**Pull request closed without merging**